python/lsst/ip/diffim/imageMapReduce.py

```diff
@@ -0,0 +1,575 @@
+            if gridStepX <= 0:
+                gridStepX = 1
+
+        if adjustGridOption == 'spacing':
+            nGridY = bbox.getWidth() / gridStepY
+            # Readjust gridStepY so that it fits perfectly in the image.
+            gridStepY = float(bbox.getHeight() - gridSizeY) / float(nGridY)
+            if gridStepY <= 0:
+                gridStepY = 1
+
+        # first "main" box at 0,0
+        bbox0 = afwGeom.Box2I(afwGeom.Point2I(bbox.getBegin()), afwGeom.Extent2I(gridSizeX, gridSizeY))
+        # first expanded box
+        bbox1 = afwGeom.Box2I(bbox0)
+        bbox1.grow(afwGeom.Extent2I(borderSizeX, borderSizeY))
+
+        self.boxes0 = []  # "main" boxes; store in task so can be extracted if needed
+        self.boxes1 = []  # "expanded" boxes
+
+        # use given centroids as centers for bounding boxes
+        if self.config.gridCentroidsX is not None and len(self.config.gridCentroidsX) > 0:
+            for i, centroidX in enumerate(self.config.gridCentroidsX):
+                centroidY = self.config.gridCentroidsY[i]
+                centroid = afwGeom.Point2D(centroidX, centroidY)
+                bb0 = afwGeom.Box2I(bbox0)
+                xoff = int(np.floor(centroid.getX())) - bb0.getWidth()//2
+                yoff = int(np.floor(centroid.getY())) - bb0.getHeight()//2
+                bb0.shift(afwGeom.Extent2I(xoff, yoff))
+                bb0.clip(bbox)
+                self.boxes0.append(bb0)
+                bb1 = afwGeom.Box2I(bbox1)
+                bb1.shift(afwGeom.Extent2I(xoff, yoff))
+                bb1.clip(bbox)
+                self.boxes1.append(bb1)
+            return self.boxes0, self.boxes1
+
+        def offsetAndClipBoxes(bbox0, bbox1, xoff, yoff, bbox):
+            """Offset the "main" (bbox0) and "expanded" (bbox1) bboxes
+            by xoff, yoff.
+
+            Clip them by the exposure's bbox.
+            """
+            xoff = int(np.floor(xoff))
+            yoff = int(np.floor(yoff))
+            bb0 = afwGeom.Box2I(bbox0)
+            bb0.shift(afwGeom.Extent2I(xoff, yoff))
+            bb0.clip(bbox)
+            bb1 = afwGeom.Box2I(bbox1)
+            bb1.shift(afwGeom.Extent2I(xoff, yoff))
+            bb1.clip(bbox)
+            if forceEvenSized:
+                bb0.grow(afwGeom.Extent2I(bb0.getWidth() % 2, bb0.getHeight() % 2))
+                bb1.grow(afwGeom.Extent2I(bb1.getWidth() % 2, bb1.getHeight() % 2))
+            return bb0, bb1
+
+        xoff = 0
+        while(xoff <= bbox.getWidth()):
+            yoff = 0
+            while(yoff <= bbox.getHeight()):
+                bb0, bb1 = offsetAndClipBoxes(bbox0, bbox1, xoff, yoff, bbox)
+                yoff += gridStepY
+                if bb0.getArea() > 0 and bb1.getArea() > 0:
+                    self.boxes0.append(bb0)
+                    self.boxes1.append(bb1)
+            xoff += gridStepX
+
+        return self.boxes0, self.boxes1
+
+    def _plotBoxes(self, exposure):
```

Comment:
I guess the question is, would you plot as part of the Task, or after the task is done?